application/src/main/java/org/togetherjava/tjbot/commands/utils/MessageUtils.java

```diff
@@ -64,19 +64,21 @@ public static String escapeMarkdown(String text) {
      * in Discord.
      *
      * @param guild the {@link Guild} that contains the command
-     * @param command the command's name
+     * @param commandInput the command's name
      * @return Formatted string for the mentioned slash command
      * @throws IllegalArgumentException when the command name doesn't match with the guild's
      *         commands
      */
-    public static RestAction<String> mentionSlashCommand(Guild guild, String command) {
-        return guild.retrieveCommands().map(commands -> {
-            Command foundCommand = commands.stream()
-                .filter(c -> c.getName().equalsIgnoreCase(command))
-                .findFirst()
-                .orElseThrow(() -> new IllegalArgumentException(
-                        "Command %s does not exist in guild %s".formatted(command, guild.getId())));
-            return String.format("</%s:%d>", command, foundCommand.getIdLong());
+    public static RestAction<String> mentionSlashCommand(Guild guild, String... commandInput) {
+        String commandWithOptions = String.join(" ", commandInput);
+        return guild.retrieveCommands().map(guildCommands -> {
+            Command guildCommand = guildCommands.stream()
+                .filter(c -> c.getName().equalsIgnoreCase(commandInput[0]))
```

Comment:
This method is easy to get wrong. Someone could think its correct to use it as `mentionSlashCommand(guild, "foo bar", "baz")` and then it would fail since it assumes the first argument to be the base name.

Also, it actually has a bug. It accesses `[0]` without length checking, nor did it require a length of > 1. I would change it to:
```java
mentionSlashCommand(Guild guild, String baseCommandName, String... subcommands)
```
or keep it a single string `String fullCommand` and then split on space for extraction urself.

Reply:
Agreed, this signature makes it easier to understand